### ISSUE TITLE
Make it worked with emqtt2.0

### ISSUE THREAD

**Issue by sudododo** [vowstar/emqttd_plugin_kafka_bridge#1]
Has anyone succeed in making this plugin worked with emqtt2.0 (emqttd-relx)? I'm trying but since I know very little about erlang I do have some difficulties. Right now I'm able to add this plugin to emqttd-relx (plugin list command can see it) and make a build, but having troubles when loading it.

**Comment by DigitalMarc** [comment]
Have you found any solution yet?

**Comment by vowstar** [comment]
@sudododo I'll update in these days

**Comment by sudododo** [comment]
@DigitalMarc Yes, you could refer to my repos. You need emqttd_relex and emq_kafka_bridge both. But right now the bridge crashes while sending the published messages. I think it's caused by the line "From = Message#mqtt_message.from". Since I know very little about erlang, I'm still trying to understand why. But connect, unconnect, and subscribe messages work perfectly.

Thank @vowstar for the initial work. It's great if you could fix the issue for publish message.

**Comment by DigitalMarc** [comment]
Thanks a lot guys. I'll wait for the next update then.

**Comment by sudododo** [comment]
The issue is due to json serialization failure for "undefined" value. In erlang, what's the proper way of doing null check?

**Comment by DigitalMarc** [comment]
I don't know much about Erlang but it seems that there is no "null" but _undefined_ atom usually takes its place.

**Comment by joe380** [comment]
Hello,
I like to ask in what stage is this plugin regarding emqttd-relx (v2.0)?
thx

**Comment by vowstar** [comment]
@joe380 On the road, and in testing
